spec/whois.md

```diff
@@ -0,0 +1,86 @@
+`did:webs` controller being an educational institution that issues "degree"
+verifiable credentials to its graduate students. A graduate from the school
+submits as part of their job application to a company a [verifiable
+presentation] derived from the verifiable credential they received from the
+school. The company receiving the presentation can verify the cryptography of
+the presentation, but can they trust the school that issued the verifiable
+credential? If the school issued the verifiable credential using its `did:webs`
+DID, the company can resolve the DID (included in the verifiable presentation
+from the student), and verify the [[ref: KEL]]. Then it can look in the DID's
+`whois` folder where it might find a couple of useful VCs or VPs with the
+`did:webs` DID as the subject. For example:
+
+- A verifiable presentation derived from a credential issued by [GLEIF]
+  containing the [LEI] of the school.
+  - Since the company knows about GLEIF, they can automate the lookup to get
+    more information about the school -- where it is registered, etc.
+- A verifiable presentation derived from a credential issued by the "Association
+  of Colleges and Universities" for the jurisdiction of the school.
+  - Since they don't know about the Association for that jurisdiction, the company can
+    repeat the process for the issuer of _that_ credential. They might (for
+    example), learn that the Association has in its `did:webs` `whois` folder a
+    verifiable credential issued by the government of the jurisdiction, and
+    the company recognizes that government.
+- A verifiable presentation derived from a credential issued by US News about
+  the school's placement on the [US News Rankings of Colleges and Universities].
+
+Such checks can all be done with a handful of HTTPS requests and the processing
+of the verifiable presentations. The result is an efficient, verifiable
+credential-based, decentralized trust registries.
+
+[GLEIF]: https://gleif.org
+[https://www.usnews.com/best-colleges/rankings/national-universities]: https://www.usnews.com/education/best-global-universities
+
+### `whois` folder Conventions
+
+The `whois` folder, if it exists, MUST be directly below the web root folder of
+the DID. The [DID URL path] to the folder MUST be `<did:webs DID>/whois`.
+
+The HTTPS path to the `whois` folder MUST be `<HTTPS URL conversion of did:webs
+identifier>/whois`. The Web Server that serves the `did:webs` documents MUST
+respond to a request for the HTTPS path to the `whois` folder with a list of the
+files within the folder so that an entity looking for information about the
+controller of the `did:webs` DID can see what verifiable credentials or
+verifiable presentations are available for review.
+
+The files in the `whois` folder must be [signed files](#signed-files), each with
+an associated [JSON Web Signature] file. The files in the folder must be either
```

Comment:
What is .jws proving? Is it that the controller of the DID intended the publication of the VC or VP? If so, wouldn't the VP prove that already -- and wouldn't the canonical way to prove intended presentation of the VC be to convert it to a VP instead of adding a .jws beside it?

Reply:
I had intended to start with verifiable presentations but added JWS files for a couple of reasons.  One is that I’ve heard that verifiable presentations are very poorly defined such that some refuse to use them. As such, I felt we need to leave open the publication of verifiable credentials, and JWSs would be needed for publishing verifiable credentials.  Next, is the “signed files” section says that for very file other than the DIDDoc, there is a JWS beside it (indicate the intention of the controller to publish the file, as you note — as well as making them tamper-evident). Since there are JWS files everywhere, why not be consistent in this folder as well.

So yes, redundent for verifiable presentations, but I think that is acceptable in this use.

Reply:
From our meeting:
There are some concerns about validating the signatures. TODO lets explore and update.